Newsletters: add issue 55 (2019-07-17)

harding opened the pull request:
TODO:

- [x] Alternatively use a bech32 sending support section being composed by @moneyball (he's currently waiting on a source, so the section included in the first draft of this PR is a backup)
- [x] Possibly remove the link to transactionfee.info.  That site is currently incorrectly reporting legacy transactions as being segwit transactions.  I pinged the site author 0xB10C about the issue via Twitter DM a couple hours ago but haven't heard back yet (I think it's the normal time for sleeping where he lives, so this is expected).  If it's not fixed by publication time, I think we should drop the link so as not to confuse readers.
- [x] Add the rest of the newsletter when it's written.  :-)

0xB10C commented on _includes/specials/bech32/18-real-fees.md:
```diff
@@ -0,0 +1,50 @@
+
+![FIXME](/img/posts/2019-07-real-cost-p2sh-p2wsh.png)
+
+For users of legacy transactions with other other transaction templates,
```
s/other other/other/

0xB10C commented:
Thank you harding for the head's up.  [transactionfee.info](https://transactionfee.info)  reported every transaction as SegWit due to a to me unknown change in an external API. This is fixed now.

fyi: [transactionfee.info](https://transactionfee.info) only shows savings for legacy -> nested SegWit. [blockstream.info](https://blockstream.info) shows nested and bech32 savings.

harding commented:
Pushed two commits.  The first fixes the typo found by @0xB10C (thanks!) and drops TransactionFee.info because it doesn't show native segwit savings as described in the post (thanks for noting that, 0xB10C!).  The second commit makes some edits, adds this week's only news item, and adds another item to the notable commits section.

jnewbery commented on _includes/specials/bech32/18-real-fees.md:
```diff
@@ -0,0 +1,50 @@
+you have an interest in that transaction, possibly significantly
+reducing your privacy.  You can privately get the approximate vbyte
+savings by aggregating the byte size of all scriptSigs in your typical
+transaction and then multiplying that number by 0.75.  When you know how
```
I think everything from "You can privately..." onward could be made clearer by turning it into step-by-step instructions, ie:

1. Use a tool to parse the transaction. Bitcoin Core comes bundled with a tool called `bitcoin-tx` that can do this for you. Run `bitcoin-tx -json <hex_serialized_tx>`
2. Add up the total size of all scriptSigs. `bitcoin-tx` outputs the hex of each scriptSig. Halve the length of this hex string to get the size of the scriptSig.
3. Multiply the size of scriptSigs that value by 0.75 to get vbytes saved.
4. Multiply the vbytes saved by feerate to get the amount of bitcoin saved.
5. Multiply the amount of bitcoin saved by the exchange rate to get savings.

bitschmidty replied:
I like the step by step instructions. (note again, for a future newsletter)

jnewbery commented on _posts/en/newsletters/2019-07-17-newsletter.md:
```diff
@@ -0,0 +1,98 @@
+  release binaries are derived solely from the Bitcoin Core source code
+  and its dependencies.  Additionally, Guix requires fewer build
+  environment dependencies and there is ongoing work to essentially
+  eliminate its need for any pre-compiled binaries in the typical build
```
I'd replace 'essentially eliminate' with 'minimize'. Even with a hex0 bootstrap (https://www.gnu.org/software/mes/manual/mes.html#The-Magical-Self_002dHosting-Hex-Assembler), you'd still need to start with a minimal binary.

harding replied:
I chose my words carefully here---that hex0 binary isn't a compiler artifact, it's handwritten assembly that simply has its comments dropped and its hex transliterated into binary.  Therefore it seems true to me that they "essentially eliminate its need for any pre-compiled binaries".  However, if you still think that's misleading, I'm happy to rewrite.

jnewbery replied:
Sounds good to me!

jnewbery commented on _posts/en/newsletters/2019-07-17-newsletter.md:
```diff
@@ -0,0 +1,98 @@
+  in the developers of Bitcoin Core and the software used to build
+  Bitcoin Core.  Although this merged PR only builds Linux binaries,
+  it is expected that support for Windows and macOS will
+  follow.
```
Perhaps link to Carl's talk at Breaking Bitcoin (https://www.youtube.com/watch?v=DKOG0BQMmmg&feature=youtu.be&t=19828) for people who want to learn more about bootstrappable and deterministic builds.

jnewbery commented on _posts/en/newsletters/2019-07-17-newsletter.md:
```diff
@@ -0,0 +1,98 @@
+  follow.
+
+- [BOLTs #607][] extends the LN specification to allow packets to contain
+  records that start with a *tag* identifying their purpose, followed by
```
s/tag/type/ ?

I've checked the PR and BOLT1, and I don't see the word 'tag' being used.

harding replied:
/me :man_facepalming:

harding commented on _includes/specials/bech32/18-real-fees.md:
```diff
@@ -0,0 +1,50 @@
+only described that in terms of vbyte and percentage savings.  In this
+bech32 sending support section, we look at the savings in real terms.
+
+The lowest practical fee is 0.00000001 BTC/vbyte.  The highest typical fees
```
Other options:

- 1e-8 BTC/vbyte
- Color fade: ![2019-07-15-13_44_48_803419173](https://user-images.githubusercontent.com/61096/61256016-d06da600-a706-11e9-8110-23e5e93d14e7.png)

0xB10C replied:
I'm curious why the `sat/vByte` unit is problematic. Most people seem to understand it and it is the shortest. 

With the `BTC/vbyte` unit a [thin space](https://en.wikipedia.org/wiki/Thin_space) might even be (somewhat) better for readability than a full space:
-  0.000 000 01 BTC/vByte |-- full space
-  0.000&thinsp;000&thinsp;01 BTC/vByte |-- thin space (`&thinsp;`)
-  0.000<sub>&thinsp;</sub>000<sub>&thinsp;</sub>01 BTC/vByte |-- thin space & subscript (`0.000<sub>&thinsp;</sub>000<sub>&thinsp;</sub>01 BTC/vByte`)

jnewbery replied:
We've discussed this internally at Optech a few times. I think Dave's main opposition to using 'satoshi' is similar to Greg's here: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-March/016709.html (that Satoshi didn't name the smallest subdivision after himself and that doing so is a bit cultish), but Dave, correct me if I'm wrong.

Like I said, we don't need to solve it this week, and we can discuss in https://github.com/bitcoinops/bitcoinops.github.io/issues/165.

harding commented on _includes/specials/bech32/18-real-fees.md:
```diff
@@ -0,0 +1,50 @@
+As of this writing, the price of bitcoin in national currencies has been
+rapidly increasing over the past few months.  We don't know whether this
+trend will continue (and it may reverse), but significant increases in
```
I'd like to leave this in.  I agree it should be obvious but, without this caveat, I feel like we're price pumping---which is something I think a technical newsletter should avoid.

harding replied:
Good compromise!  Pushed a commit that makes this edit, thanks.

bitschmidty replied:
Marking this conversation as resolved. Note that this segment was bumped in favor of the bech32 support issues segment this week.

harding commented:
Pushed two commits.  The first addresses feedback from @jnewbery (thanks!) minus that which I left comments on above.  The second replaces the bech32 section with the excellent one written by @moneyball.  I left the previous section in the PR and did not rename it so that all comments remain valid; I suggest we try to resolve any open threads on it, merge it, and then I'll rename it and use it in next week's PR.

jnewbery commented:
ACK 7336b37eaf06a30aa017f9fedce2ec51f77dafde

Thanks @harding !

bitschmidty commented:
@harding comes through, never late
even when outside the lower 48

@moneyball explains what services do
when deciding to receive to bech32

@jnewbery with another review
and @0xB10C with one too

(thanks all!)
